pkg/buildscript/internal/raw/mutations.go

```diff
@@ -0,0 +1,171 @@
+package raw
+
+import (
+	"github.com/go-openapi/strfmt"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/rtutils/ptr"
+	"github.com/ActiveState/cli/pkg/platform/api/buildplanner/types"
+)
+
+const requirementRevisionKey = "revision"
+
+func (r *Raw) UpdateRequirement(operation types.Operation, requirement types.Requirement) error {
```

Comment:
I've decided to move the raw struct into the main package, but as a private member. I tried doing what you suggested with a clear separation of responsibilities, but I ran into two issues:

1. Unmarshaling build expressions into a raw structure requires a bit of interpretation after the fact (transforming requirement object notation into function notation). Since this mutation should be done in build scripts directly, it crosses the separation boundary. Requiring the mutation to happen outside of raw's unmarshaling puts raw into an "incomplete state" (for lack of a better word) that needs further work. Forgetting to do that work will result in head-scratching bugs.
2. Since build scripts and build expressions differ slightly in structure (e.g. requirement functions vs. objects), there's an open question of what exactly raw is supposed to marshal. What I mean is that suppose we are going to marshal a build script into a Platform build expression. Raw stores requirements as functions, but we need to convert those functions to objects for the build expression. Is raw's marshal supposed to do this? Or is the build script supposed to convert functions to objects and then have raw marshal those objects? By moving raw into the main package as a private member, I can sidestep this issue entirely and have just one marshal function for the build script, which does the right thing.